src/common/snippets/src/pass/common_optimizations.cpp

```diff
@@ -68,82 +82,87 @@ void CommonOptimizations::SplitDimensionM(const std::shared_ptr<ov::snippets::op
 
     const auto mm_shape = matmul0->get_shape();
     const auto m_dim = get_dim_M(mm_shape);  // M
-    const auto n_dim = mm_shape.back(); // N
+    const auto batch_dim =
+        std::accumulate(mm_shape.rbegin() + 2, mm_shape.rend(), size_t(1), std::multiplies<size_t>());  // B (batch)
     // [113745] Heurestic is equal to double block size.
     // When this optimization will be moved into Subgraph and blocking param will be implemented as dependents of shapes,
     // need to implement common way (for all backends) to calculate optimal value of M dimension
     const auto optimal_m_dim = 32 * 2;
-    const auto optimal_parallelism_work_amount = minimal_concurrency;
-    if (m_dim <= optimal_m_dim)
+    const auto optimal_parallelism_work_amount = concurrency;
+    // We skip optimization
+    //  - if the current M dimension is not enough for splitting
+    //  - if the current batch is optimal for concurrency
+    if (m_dim <= optimal_m_dim || batch_dim % optimal_parallelism_work_amount == 0)
         return;
 
-    const auto batch_dim =
-        std::accumulate(mm_shape.rbegin() + 2, mm_shape.rend(), size_t(1), std::multiplies<size_t>());  // B (batch)
     size_t batch_m_dim = 1;
     size_t new_m_dim = m_dim;
 
+    auto is_optimized = [&](size_t batch_m_dim, size_t new_m_dim) {
+        return batch_m_dim > 1 && new_m_dim >= static_cast<size_t>(optimal_m_dim);
```

Comment:
To add a point to the prev comment: `optimal_m_dim` defines our stopping criterion, so the choice of its value could affect performance significantly. We probably need to be more careful with it.

Reply:
The function `is_optimized` was just moved in the file. I didn't change this logic and didn't break the previous logic. I think if we have more shapes, more models, we would be able to adjust the logic more carefully or flexible. But we have what we have. Thanks!

Reply:
Discussed offline. Agreed to allow for splitting the only M block (remove `optimal_m_dim`). Need to perform validation.

Reply:
As we discussed offline, I removed `optimal_m_dim` and validated on model set. The results are attached to the ticket